engines/base/src/test/java/org/alfresco/transform/base/LivenessReadinessProbeTest.java

```diff
@@ -0,0 +1,107 @@
+package org.alfresco.transform.base;
+
+import org.junit.jupiter.api.Test;
+
+import org.springframework.core.io.ClassPathResource;
+
+import org.springframework.http.client.MultipartBodyBuilder;
+import org.springframework.test.web.reactive.server.WebTestClient;
+import org.testcontainers.containers.GenericContainer;
+import org.testcontainers.containers.wait.strategy.Wait;
+
+import java.net.URISyntaxException;
+
+import java.util.Objects;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+
+import static org.springframework.http.HttpStatus.OK;
+import static org.springframework.http.HttpStatus.TOO_MANY_REQUESTS;
+
+import org.springframework.web.reactive.function.client.WebClient;
+
+public abstract class LivenessReadinessProbeTest
+{
+    @Test
+    public void readinessShouldReturnAn429ErrorAfterReachingMaxTransforms() throws URISyntaxException
+    {
+        final ImagesForTests testData = getImageForTest();
+
+        try (final var env = createEnv(testData.image))
+        {
+            env.start();
+            var url = "http://localhost:" + env.getFirstMappedPort();
+
+            int max_transforms = 11;
+            for (int i = 0; i<max_transforms; i++) {
+                sendTransformRequest(url, testData.sourceMimetype, testData.targetMimetype, testData.filename);
```

Comment:
Would be good to add an assertion that probe is successful before sending transformation request

Reply:
Added an assertion before sending each request